codebuild/bin/test_install_shared_and_static.sh

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env bash
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+set -e
+
+usage() {
+    echo "test_install_shared_and_static.sh build_dir"
+    echo "tests that installed s2n-config.cmake chooses appropriately between shared and static"
+    exit 1
+}
+
+if [ "$#" -ne 1 ]; then
+    usage
+fi
+
+WORK_DIR=$1
+
+source codebuild/bin/s2n_setup_env.sh
+
+source codebuild/bin/jobs.sh
+export CMAKE_BUILD_PARALLEL_LEVEL=$JOBS
+
+COMMON_S2N_BUILD_ARGS=(-H. -DCMAKE_PREFIX_PATH=$LIBCRYPTO_ROOT -DBUILD_TESTING=OFF)
+
+# create installation dir with libs2n.so
+if [ ! -d $WORK_DIR/s2n-install-shared ]; then
+    (set -x; cmake -B$WORK_DIR/s2n-build-shared -DCMAKE_INSTALL_PREFIX=$WORK_DIR/s2n-install-shared -DBUILD_SHARED_LIBS=ON ${COMMON_S2N_BUILD_ARGS[@]})
+    (set -x; cmake --build $WORK_DIR/s2n-build-shared --target install)
```

Comment:
The `--target install` seems to cause cmake to link against the system libcrypto (openssl 1.1 on ubuntu18). Setting  `S2N_LIBCRYPTO` to anything else causes this test to fail (built vs linked).  Might be good to check the S2N_LIBCRYPTO/LIBCRYPTO_ROOT and bail out with something like `only supported on openssl-1.1.1 in CI`

Reply:
Thanks for the clue that `--target install` was doing something fishy.

Turns out the install step strips off the RPATH/RUNPATH that lets in find libcrypto.so in non-standard locations.

I pushed a fix that sets LD_LIBRARY_PATH to point at libcrypto.so's location. That way an installed s2n can still use the same version of libcrypto.so it was built against, even if libcrypto.so isn't in an offical system lib directory